ingestion/src/metadata/ingestion/ometa/mixins/tag_mixin.py

```diff
@@ -0,0 +1,181 @@
+"""
+Mixin class containing entity tag specific methods
+
+To be used be OpenMetadata
+"""
+import logging
+from typing import List, Optional, Type, TypeVar
+
+from pydantic import BaseModel
+
+from metadata.generated.schema.entity.tags.tagCategory import Tag, TagCategory
+from metadata.ingestion.ometa.client import APIError
+
+T = TypeVar("T", bound=BaseModel)  # pylint: disable=invalid-name
+logger = logging.getLogger(__name__)
+
+
+class OMetaTagMixin:
+    """
+    OpenMetadata API methods related to entity tag.
+
+    To be inherited by OpenMetadata
+    """
+
+    def get_list_tag_categories(
+        self, entity: Type[T], fields: Optional[List[str]] = None
+    ) -> Optional[List[T]]:
+        """Get list of TagCategory pydantic model
+
+        Args:
+            entity: entity class model
+            fields (List): list of fields to pass with the request
+        """
+
+        fields_str = "?fields=" + ",".join(fields) if fields else ""
+        try:
+            resp = self.client.get(f"{self.get_suffix(entity)}/{fields_str}")
+            return [entity(**tag_cat) for tag_cat in resp.get("data")]
+        except APIError as err:
+            logger.error(f"GET {entity.__name__}. Error {err.status_code} - {err}")
+            return None
+
+    def post_create_tag_category(self, tag_category_body: TagCategory):
```

Comment:
Hi @TeddyCr, amazing job here 🚀 

May I suggest a little naming change? It might reduce some verbosity and make the code easier to read if we remove unnecessary prefixes such as `post` or `put` in the function names. Instead of `post_create_tag_category`, we could stick to `create_tag_category` and leave the underlying implementation a bit more hidden.

I'd do the same for the `list` methods, and just have `list_tag_categories`, while in single `get`s, I think the name is already on point with `get_tag_category`.

WDYT?

Reply:
Yep that makes sense.